dispatcher.go

```diff
@@ -32,6 +34,11 @@ type EventStream[T any] interface {
 	// Stream returns access to the event T channel where events will arrive.
 	Stream() <-chan T
 
+	// SyncStream returns a channel that receives SyncEvent[T] instances containing the
+	// event T payload and a channel to signal when the event is processed. This channel
+	// *ONLY* receives events that are dispatched with DispatchSync.
```

Comment:
> // This channel *ONLY* receives events that are dispatched with DispatchSync.

This isn't true - it was initially, but I decided it made the API way too confusing, so now `SyncStream()` receives events dispatched using Dispatch or DispatchSync, but only blocks on events dispatched with DispatchSync.

Reply:
The README section was awesome, by the way!

Reply:
Updated godoc text for this to match README